### ISSUE TITLE
Cannot load HTTP vector sources

### ISSUE THREAD

**Issue by keum** [maputnik/editor#177]
When I tried to load the data using https://maputnik.github.io/editor/#10.82/47.6967/-122.2273
this site, it doesn't work. 
But when I do load onto this maputnik http://editor.openmaptiles.org/#10.75/47.6140/-122.3843
it works fine. 
Can someone can explain the difference between maputnik that's running on github.io and another one running off openmaptiles folks?
I'll be greatly appreciated for your help.

**Comment by lukasmartinelli** [comment]
Wondering whether it has to do with HTTP/S and mixed content.

2017-10-16 12:28 GMT-04:00 Peter Keum <notifications@github.com>:

**Comment by orangemug** [comment]
@keum if you use https (https://editor.openmaptiles.org/#10.75/47.6140/-122.3843) do you get the same issue?

**Comment by keum** [comment]
@orangemug no, openmaptile version of maptunik works fine and renders the data.  I think what @lukasmartinelli is thinking that http/s might has to do with it. Openmaptile.org version  doesn't have https..just http.  I'm trying to add source from my DigitalOcean vector tile that I've created here. http://vectortile.pkgeo.com:8080/data/wabldg.json.  I think i see the problem, github's starts w/ HTTPS. any suggestion on getting around this so wabldg.json can be added to both version of maputnik? thanks as always.

**Comment by keum** [comment]
I see why it's doing it. I need to change my tileserver to serve https instead of http. I' think you can close this issue.

**Comment by lukasmartinelli** [comment]
@orangemug I imagine the majority of people will serve their tiles via HTTP... and then they cannot use maputnik.github.io to style them quickly - cause it is HTTP.

I researched you cannot disable HTTPS enforcement on GitHub pages 😢 .
So we either promote more to download maputnik locally (since you can literally just download a zip file and open with browser) or provide a HTTP alternative (host it on maputnik.com) again.

**Comment by keum** [comment]
@lukasmartinelli there is two version for editor.openmaptiles.org
http://editor.openmaptiles.org/#8.89/47.6878/-122.4870
https://editor.openmaptiles.org/#10.73/47.5711/-122.4048
Looks like Klokan Tech is exposing both http and https...using http version, i can add vector data. what is relation between editor.openmaptiles.org maptunik & maputink.github.io ? are two are same code? thanks for the input.

**Comment by orangemug** [comment]
@lukasmartinelli so I'm a fan of using `maputnik.com` I think having our own domain would make things a lot easier in the long run. I've got a number of things I'd like to work on that might have a server component, for example full github integration using something like https://github.com/prose/gatekeeper for auth. I think if the project continues to grow having our own domain again would be inevitable.

As for the http/https issue we really should be encouraging people to use https, for security reasons. Allowing http as an option only leads users into a situation where they won't be able to use https sources later on. I think this will cause confusion for users. The exported styles will also work in `bl.ocks.org` and github pages, both of which don't support non SSL.

We should however show a better error if users try to enter `http://` into the sources panel. With a nice message to say something like

> Non SSL (http://) sources are not supported, this is due to the "Same-origin policy" of the browser, read more [here](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)

...yeah this message could be a lot better, but you get the idea

Thoughts?

**Comment by lukasmartinelli** [comment]
> As for the http/https issue we really should be encouraging people to use https, for security reasons. Allowing http as an option only leads users into a situation where they won't be able to use https sources later on. I think this will cause confusion for users. The exported styles will also work in bl.ocks.org and github pages, both of which don't support non SSL.

For serving tiles online.
But what about people who just use it to style some tiles lying around locally.

We can resurrect maputnik.com again (reverting the maputnik.github.io PR), the CNAME is already still pointing to github.

Force push master from https://github.com/maputnik/maputnik.github.io to https://github.com/maputnik/maputnik.com and add CNAME file.

**Comment by keum** [comment]
thanks @lukasmartinelli I agree. I think for local development, you'll just need to use locally installed Maputnik.  I'm satisfied w/ options so you're free to close the tix if you would like.

**Comment by orangemug** [comment]
> But what about people who just use it to style some tiles lying around locally.

@lukasmartinelli good point. Allowing both http and https sounds good to me at this point then.

In future I may look to create a standalone static file server that supports https. I have some experience of this with https://github.com/orangemug/manage-hosts, if anyone is interested this branch https://github.com/orangemug/manage-hosts/compare/feature/https adds self signed certs on OSX (messy but works)

**Comment by keum** [comment]
I just finished maptime seattle last night and used https://github.com version. it's little cumbersome but when you load the source data, I needed to disabled the security on the browser in Firefox to render the http source vector data. 
I also used the same method to see preview in bl.ocks.org as well...all went fantastically smooth and students loved using Maputnik  to see vector tiles being stylized and able to view the web map w/ any style they wanted!  I would like to close the comment on this issue.
